cli/run.go

```diff
@@ -140,16 +169,19 @@ func setupAgentConfig() (*configs.AgentConfig, error) {
 }
 
 type WorkflowRunner struct {
-	config      RunConfig
-	
+	config RunConfig
+
 	// agentConfig is only non-nil if the CLI is configured to run in agent mode
-	agentConfig *configs.AgentConfig
+	agentConfig   *configs.AgentConfig
+	dockerManager DockerManager
 }
 
 func NewWorkflowRunner(config RunConfig, agentConfig *configs.AgentConfig) WorkflowRunner {
+	_, stepSecretValues := tools.GetSecretKeysAndValues(config.Secrets)
 	return WorkflowRunner{
-		config:      config,
-		agentConfig: agentConfig,
+		config:        config,
+		dockerManager: docker.NewContainerManager(log.NewLogger(log.GetGlobalLoggerOpts()), stepSecretValues),
```

Comment:
Does this connect to the docker daemon instantly or just lazily? I'm asking because this is a niche feature and won't be used in every execution, we shouldn't do any heavy lifting here if not needed.

Reply:
See the `agentConfig` above for a similar example: it's an optional field that's not initialized when it's irrelevant for the execution.

Reply:
This is just a simple API client, it should not cause any significant overhead in initialization. We tested and it can initialize without an error on both of our Linux and mac stacks.

Reply:
Got it, thank you!